gamemodes/terrortown/gamemode/shared/sh_equip_items.lua

```diff
@@ -811,47 +816,24 @@ function AddToShopFallback(fallback, subrole, eq)
 	end
 end
 
-local function InitDefaultEquipmentForRole(roleData)
-	-- set default equipment tables
-	local itms = items.GetList()
-	local sweps = weapons.GetList()
-
-	-- TRAITOR
-	local tbl = {}
-
-	-- find buyable items to load info from
-	for i = 1, #itms do
-		local v = itms[i]
-
-		if not v or v.Doublicated or not v.CanBuy or not v.CanBuy[roleData.index] then continue end
-
-		local data = v.EquipMenuData or {}
-
-		local base = GetEquipmentBase(data, v)
-		if not base then continue end
-
-		tbl[#tbl + 1] = base
-	end
-
-	-- find buyable weapons to load info from
-	for i = 1, #sweps do
-		local v = sweps[i]
-
-		if not v or v.Doublicated or not v.CanBuy or not v.CanBuy[roleData.index] then continue end
+---
+-- Initializes the default equipment for a role
+-- @internal
+-- @realm shared
+local function InitDefaultEquipmentForRole(roleData, eq)
+	local tbl = roleData.fallbackTable or {}
 
-		local data = v.EquipMenuData or {}
+	-- is a buyable equipment to load info from
+	if not eq or eq.Doublicated or not eq.CanBuy or not eq.CanBuy[roleData.index] then return end
 
-		local base = GetEquipmentBase(data, v)
-		if not base then continue end
+	local base = GetEquipmentBase(eq)
+	if not base then return end
 
-		tbl[#tbl + 1] = base
-	end
+	tbl[#tbl + 1] = base
 
-	-- mark custom items
-	for _, i in pairs(tbl) do
-		if i and i.id then
-			i.custom = not table.HasValue(DefaultEquipment[roleData.index], i.id) -- TODO
-		end
+	-- mark custom equipment
+	if base and base.id then
+		base.custom = not table.HasValue(DefaultEquipment[roleData.index], base.id) -- TODO
```

Comment:
has anyone here an idea what this todo means? It should be removed and put into trello

Reply:
https://github.com/TTT-2/TTT2/blob/master/gamemodes/terrortown/gamemode/shared/sh_init.lua#L126-L127

Seems to come from here, there are several such todos, that apparently check for default roles.
![grafik](https://user-images.githubusercontent.com/72046466/128474782-39e520ad-3a15-4f9d-b093-af15e8a6a465.png)

Reply:
ahh, hmm @Alf21 is this something that has to be done? Or can it be removed?

Reply:
custom marks the equipment as a custom created one (star or something similar). That doesn't make sense, as nearly every item is a custom one. The user has no advantage from it. That's why I added a TODO to think about it and use this already integrated feature in another way

Reply:
Ah, I see
this is something that will be changed with the planned shop UI rework. Thanks! I'll write something in Trello so you @ZenBre4ker can remove the todo here